core/src/main/kotlin/supplier/RestEntitySupplier.kt

```diff
@@ -440,11 +440,11 @@ public class RestEntitySupplier(public val kord: Kord) : EntitySupplier {
         applicationId: Snowflake,
         interactionToken: String,
         messageId: Snowflake,
-    ): PublicFollowupMessage? = catchNotFound {
```

Comment:
Why is it that the types here changed

Reply:
When the endpoint returns a message with the ephemeral flag it shouldn't be a `PublicFollowupMessage`.

Reply:
so we are going to use flags to determine ephemeral and public?

Reply:
That's exactly what the `FollowupMessage()` function does. The `edit` extension function for `FollowupMessageBehavior` already did this before.

Reply:
Or is there a downside to this that I'm missing?

Reply:
well the FollowupMessage is not a concrete types so the people may need to cast; maybe provide convenience methods as well.

Reply:
What would they look like? `asPublicFollowupMessage()`? And would they return `null` or throw on wrong type?

Reply:
no, make two distinct getters without this `asX` pattern here; because you can then call this function on a ephemeral function and make it public? is this of a value to us?

Reply:
Can you give me an example?

Reply:
Yeah that's what I was trying to imply; have there been any talks about supporting it? (deleting an ephemeral)

Reply:
This comment did somewhat imply it: https://github.com/discord/discord-api-docs/pull/3866#issuecomment-930408139

Reply:
Good news; then this issue is resolved.

Reply:
is it worth merging knowing that this will change?
I think you can just remove the behavioral distinction rather than re adjusting later

Reply:
Not entirely; the type distinction between the Ephemeral and Public is still required; but the behavior will be the same

Reply:
you know Discord have the ephmeral flag; we don't usually compare by types and flags for entites 
e.g DmChannel and TextChannel; they have their own types
Ephemeral and Public are parts of the concepts of interactions; also it's nice to express the distinction as an `is` rather than ==; it reads better and being a firm type makes it good.

On the other hand; your PR seems to have changed some of the parts that are going to change in V10 (which are not that hard to migrate to  https://github.com/discord/discord-api-docs/discussions/4510

not sure abotu the attachment patch thing yet tho

Reply:
Ok makes sense to have separate types for this.

So the thing you suggest now is to migrate all methods on followup messages to `FollowupMessageBehavior` (and including a warning that it might not work for ephemerals)?


> On the other hand; your PR seems to have changed some of the parts that are going to change in V10 (which are not that hard to migrate to https://github.com/discord/discord-api-docs/discussions/4510

Not sure which changes exactly you are referring to here.

Reply:
> [ALL VERSIONS] application.summary now returns an empty string. This field will be removed in v11

although we need a separate PR for this

Reply:
Oh, this could be in scope for https://github.com/kordlib/kord/pull/530